nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/CompoundDocumentServiceImpl.java

```diff
@@ -43,20 +46,34 @@
     protected static final String XP_IGNORED_FILES = "ignoredFiles";
 
     @Override
-    public DocumentModel createCompoundDocument(CoreSession session, String parent, Blob archive) {
+    public DocumentModel createCompoundDocument(CoreSession session, String parent, Blob archive)
+            throws OperationException {
         String compoundDocName = FilenameUtils.removeExtension(archive.getFilename());
-        DocumentModel compoundDoc = session.createDocumentModel(parent, compoundDocName, COMPOUND_DOCTYPE);
+        String compoundType = runScript(COMPOUND_DOCTYPE_DETECTION_OPERATION);
+        DocumentModel compoundDoc = session.createDocumentModel(parent, compoundDocName, compoundType);
         compoundDoc = session.createDocument(compoundDoc);
         createCompoundDocument(compoundDoc, archive);
         return compoundDoc;
     }
 
+    protected String runScript(String scriptId) throws OperationException {
+        AutomationService automationService = Framework.getService(AutomationService.class);
+        OperationContext ctx = new OperationContext();
+        return (String) automationService.run(ctx, scriptId);
+    }
```

Comment:
The current implementation can not provide the need described in the JIRA issue as there're no input/parameters given to the automation.

Furthermore, `OperationContext` is a closable resource.